pkg/gather/aws/aws.go

```diff
@@ -0,0 +1,236 @@
+		return err
+	}
+
+	var errs []error
+	var files []string
+	for _, instance := range instances {
+		filePath, err := g.downloadConsoleOutput(ctx, ec2Client, instance, filePathDir)
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			files = append(files, filePath)
+		}
+	}
+
+	if len(files) > 0 {
+		err := gather.CreateArchive(files, g.serialLogBundle)
+		if err != nil {
+			errs = append(errs, errors.Wrap(err, "failed to create archive"))
+		}
+	}
+
+	if err := gather.DeleteArchiveDirectory(filePathDir); err != nil {
+		// Note: cleanup is best effort, it shouldn't fail the gather
+		g.logger.Debugf("failed to remove archive directory: %v", err)
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+// findEC2Instances returns the EC2 instances with tags that satisfy the filters.
+func (g *Gather) findEC2Instances(ctx context.Context, ec2Client *ec2.EC2) ([]*ec2.Instance, error) {
+	if ec2Client.Config.Region == nil {
+		return nil, errors.New("EC2 client does not have region configured")
+	}
+
+	var instances []*ec2.Instance
+	for _, filter := range g.filters {
+		g.logger.Debugf("search for matching instances by tag in %s matching %#+v", *ec2Client.Config.Region, filter)
+		instanceFilters := make([]*ec2.Filter, 0, len(g.filters))
+		for key, value := range filter {
+			instanceFilters = append(instanceFilters, &ec2.Filter{
+				Name:   aws.String("tag:" + key),
+				Values: []*string{aws.String(value)},
+			})
+		}
+
+		err := ec2Client.DescribeInstancesPagesWithContext(
+			ctx,
+			&ec2.DescribeInstancesInput{Filters: instanceFilters},
+			func(results *ec2.DescribeInstancesOutput, lastPage bool) bool {
+				for _, reservation := range results.Reservations {
+					if reservation.OwnerId == nil {
+						continue
+					}
+
+					for _, instance := range reservation.Instances {
+						if instance.InstanceId != nil {
+							instances = append(instances, instance)
+						}
+					}
+				}
+				return !lastPage
+			},
+		)
+		if err != nil {
+			err = errors.Wrap(err, "get ec2 instances")
+			return instances, err
+		}
+	}
+
+	return instances, nil
+}
+
+func (g *Gather) downloadConsoleOutput(ctx context.Context, ec2Client *ec2.EC2, instance *ec2.Instance, filePathDir string) (string, error) {
+	logger := g.logger.WithField("instance", *instance.InstanceId)
+
+	input := &ec2.GetConsoleOutputInput{
+		InstanceId: aws.String(*instance.InstanceId),
+	}
+	result, err := ec2Client.GetConsoleOutputWithContext(ctx, input)
+	if err != nil {
+		// Cast err to awserr.Error to get the Message from an error.
+		if aerr, ok := err.(awserr.Error); ok {
+			logger.Errorln(aerr.Error())
+		}
+		return "", err
+	}
+
+	instanceName := *result.InstanceId
+	for _, tags := range instance.Tags {
+		if strings.EqualFold(*tags.Key, "Name") {
+			instanceName = *tags.Value
+		}
+	}
+
+	logger.Debugf("attemping to download console logs for %s", instanceName)
+	filePath, err := g.saveToFile(instanceName, *result.Output, filePathDir)
```

Comment:
It just happened to me now that `result.Output` is `nil`. I'll fix this so we don't dereference a `nil` pointer.